marathon-bigip-ctlr.py

```diff
@@ -600,53 +750,84 @@ def create_config_marathon(bigip, apps):
         frontend_name = frontend_name.replace('/', '_')
-        f5_service['name'] = frontend_name
+
         if app.bindAddr:
             logger.debug("Frontend at %s:%d with backend %s", app.bindAddr,
                          app.servicePort, backend)
 
+        # pool members
+        members = []
+        key_func = attrgetter('host', 'port')
+        for backendServer in sorted(app.backends, key=key_func):
+            logger.debug("Found backend server at %s:%d for app %s",
+                         backendServer.host, backendServer.port, app.appId)
+
+            # Resolve backendServer hostname to IP address
+            ipv4 = resolve_ip(backendServer.host)
+
+            if ipv4 is not None:
+                member = {
+                    'address': ipv4,
+                    'port': backendServer.port,
+                    'state': 'user-up',
+                    'session': 'user-enabled'
+                }
+                members.append(member)
+            else:
+                logger.warning("Could not resolve ip for host %s, "
+                               "ignoring this backend", backendServer.host)
+
         if app.iapp:
             # Translate from the internal properties we set on app to the
-            # naming expected by the common f5_service['iapp']
+            # naming expected by the iapp.
             # Only set properties that are actually present.
-            #
-            # tableName would be better as poolMemberTableName but it is
-            # required to be tableName to match the tableName produced by
-            # the k8s-bigip-ctlr
-            f5_service['iapp'] = {}
+            cfg = {}
             for k, v in {'template': 'iapp',
                          'tableName': 'iappPoolMemberTableName',
                          'poolMemberTable': 'iappPoolMemberTable',
                          'tables': 'iappTables',
                          'variables': 'iappVariables',
                          'options': 'iappOptions'}.iteritems():
                 if hasattr(app, v):
-                    f5_service['iapp'][k] = getattr(app, v)
+                    cfg[k] = getattr(app, v)
 
             # Decode the tables
             for key in app.iappTables:
-                f5_service['iapp']['tables'][key] = \
+                cfg['tables'][key] = \
                     json.loads(app.iappTables[key])
-        else:
-            f5_service['virtual'] = {}
-            f5_service['pool'] = {}
-            f5_service['health'] = []
 
+            # format the variables and tables per the schema
+            iapp_def = iapp_build_definition(cfg, members)
+
+            iapp = {
+                'name': frontend_name,
+                'template': cfg['template'],
+                'variables': iapp_def['variables'],
+                'tables': iapp_def['tables'],
+                'options': cfg['options']
+            }
+
+            services['iapps'].append(iapp)
+        else:
+            monitors = []
```

Comment:
Need to keep a list of monitors specific for a service, which are later added to the service's pool.